src/arim/core.py

```diff
@@ -1344,19 +1356,21 @@ class BlockInImmersion(ExaminationObject):
     material : Material
         Alias for block_material
     couplant_material : Material
-    frontwall : OrientedPoints
-    backwall : OrientedPoints or None
+    walls : list[OrientedPoints]
+    wall_idxs_for_imaging : list[int]
```

Comment:
I would prefer `ExaminationObject` to remain agnostic to the details of imaging algorithm. Which wall to use for imaging should be in my opinion a parameter of `arim.models.block_in_contact.make_views` or `arim.models.block_in_immersion.make_views`.

Reply:
This is a good point. I've implemented this by replacing the `max_number_of_reflection` parameter with a `walls_for_imaging` parameter which takes a list of strings, which are the keys in `examination_object.walls` which are used. The length of this list is therefore equal to `max_number_of_reflection`.